client/rest-high-level/src/test/java/org/elasticsearch/client/IngestClientIT.java

```diff
@@ -80,4 +89,71 @@ public void testDeletePipeline() throws IOException {
+        builder.startObject();
+        {
+            builder.field("pipeline");
+            buildRandomXContentPipeline(builder);
+            builder.startArray("docs");
+            {
+                for (int i = 0; i < numDocs; i++) {
+                    builder.startObject();
+                    IngestDocument document = RandomDocumentPicks.randomIngestDocument(random());
+                    Map<IngestDocument.MetaData, Object> metadataMap = document.extractMetadata();
+                    for (Map.Entry<IngestDocument.MetaData, Object> metadata : metadataMap.entrySet()) {
+                        if (metadata.getValue() != null) {
+                            if (metadata.getKey().equals(IngestDocument.MetaData.VERSION)) {
+                                builder.field(metadata.getKey().getFieldName(), (long)metadata.getValue());
+                            } else {
+                                builder.field(metadata.getKey().getFieldName(), metadata.getValue().toString());
+                            }
+                        }
+                    }
+                    document.setFieldValue("rank", Integer.toString(randomInt()));
+                    builder.field("_source", document.getSourceAndMetadata());
+                    builder.endObject();
+                }
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+
+        SimulatePipelineRequest request = new SimulatePipelineRequest(
+            BytesReference.bytes(builder),
+            builder.contentType()
+        );
+        request.setVerbose(isVerbose);
+
+        SimulatePipelineResponse simulatePipelineResponse =
+            execute(request, highLevelClient().ingest()::simulatePipeline, highLevelClient().ingest()::simulatePipelineAsync);
+
+        for (SimulateDocumentResult result: simulatePipelineResponse.getResults()) {
+            if (isVerbose) {
+                assertTrue(result instanceof SimulateDocumentVerboseResult);
+                SimulateDocumentVerboseResult verboseResult = (SimulateDocumentVerboseResult)result;
+                assertTrue(verboseResult.getProcessorResults().size() > 0);
+                assertEquals(
+                    verboseResult.getProcessorResults().get(0).getIngestDocument()
+                        .getFieldValue("foo", String.class),
+                    "bar"
+                );
+            } else {
+                assertTrue(result instanceof SimulateDocumentBaseResult);
+                SimulateDocumentBaseResult baseResult = (SimulateDocumentBaseResult)result;
+                assertNotNull(baseResult.getIngestDocument());
+                assertEquals(
+                    baseResult.getIngestDocument().getFieldValue("foo", String.class),
+                    "bar"
+                );
+                assertNotNull(
+                    baseResult.getIngestDocument().getFieldValue("rank", Integer.class)
+                );
+            }
+        }
+    }
```

Comment:
I think that being this an integration test, we should randomize much less, if at all. We can have a fixed request and check that the response is expected, let's leave the randomizations for the unit tests.

Reply:
can you also the test the case where you get back an error?

Reply:
shall we test the failure separately?